openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -181,6 +181,40 @@ def test_requirements_to_disable(self):
         requirements_to_disabled = _get_requirements_to_disable(old_requirements, requirements)
         self.assertEqual(len(requirements_to_disabled), 0)
 
+    def test_set_credit_requirement_status(self):
+        self.add_credit_course()
+        requirements = [
+            {
+                "namespace": "grade",
+                "name": "grade",
+                "display_name": "Grade",
+                "criteria": {
+                    "min_grade": 0.8
+                }
+            },
+            {
+                "namespace": "reverification",
+                "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
+                "display_name": "Assessment 1",
+                "criteria": {}
+            }
+        ]
+
+        set_credit_requirements(self.course_key, requirements)
+        course_requirements = CreditRequirement.get_course_requirements(self.course_key)
+        self.assertEqual(len(course_requirements), 2)
+
+        requirement = CreditRequirement.get_course_requirement(self.course_key, "grade", "grade")
```

Comment:
I have made these api functions as per suggestion of @wedaly to not use model methods directly in other apps. https://github.com/edx/edx-platform/pull/8228#discussion_r31355073